### DISCUSSION TITLE
Add Raspberry Pi 5 support [21000 sats⚡by @KevinR4v ]

### DISCUSSION THREAD

**Discussion by kravens** [raspiblitz/raspiblitz#4236] [Bounty Ideas]
I just received the Raspberry Pi5, which has a few crucial differences from the previous Pi4 board and requires Debian 12 based Raspberry Pi OS.
One that I ran into is the default external SSD location, which is now: /media/pi/hdd instead of /mnt/hdd

>/home/admin/config.scripts/network.txindex.sh: line 10: /mnt/hdd/raspiblitz.conf: No such file or directory
grep: /mnt/hdd//.conf: No such file or directory
tee: /mnt/hdd//.conf: No such file or directory
grep: /mnt/hdd//.conf: No such file or directory
/home/admin/config.scripts/network.txindex.sh: line 40: [: -eq: unary operator expected
tail: cannot open '/mnt/hdd//debug.log' for reading: No such file or directory
tail: cannot open '/mnt/hdd//debug.log' for reading: No such file or directory

We should find all the changes required to make RaspiBlitz work smoothly again on this new and faster Pi 5!
I'll pay 21k sats bounty to the person making a RaspiBlitz 1.10 (or future) image compatible with the Pi 5 ⚡

**Comment by kravens** [comment]
Another issue I'm running into is the constantly changing ssh fingerprint each reboot:

>@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ED25519 key sent by the remote host is
SHA256:C0RNQW8qw76VZjUtschHPgXngSEK57YRyGPMvazOgeE.
Please contact your system administrator.

**Comment by openoms** [comment]
Thank you for documenting this and offering a bounty!
Got these noted from the release notes so far: https://github.com/raspiblitz/raspiblitz/issues/4215
The new base image is being tested in github actions: https://github.com/raspiblitz/raspiblitz/pull/4238
Once working it should be compatible with both the RPi4 and RPi5.

**Comment by kravens** [comment]
>*** DISABLE BLUETOOTH ***
># Raspiblitz
>dtoverlay=pi3-disable-bt
dtoverlay=disable-bt
Failed to disable unit: Unit file bluetooth.service does not exist.
Failed to disable unit: Unit file hciuart.service does not exist.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done

---

>*** CONFIG ***
>autostart LCD already in /home/pi/.bashrc
># Optimizing log files: rotate daily max 100M, keep 4 days & compress old
>
>/var/log/syslog
/var/log/mail.info
/var/log/mail.warn
/var/log/mail.err
/var/log/mail.log
/var/log/daemon.log
/var/log/kern.log
/var/log/auth.log
/var/log/user.log
/var/log/lpr.log
/var/log/cron.log
/var/log/debug
/var/log/messages
{
  rotate 4
  size 100M
  missingok
  compress
  delaycompress
  copytruncate
  sharedscripts
  postrotate
    service logrotate restart
  endscript
}

>Job for logrotate.service failed because the control process exited with error code.
See "systemctl status logrotate.service" and "journalctl -xeu logrotate.service" for details.
Failed to restart rsyslog.service: Unit rsyslog.service not found.

---

 >     building 'cchardet._cchardet' extension
      creating build/temp.linux-aarch64-cpython-311
      creating build/temp.linux-aarch64-cpython-311/src
      creating build/temp.linux-aarch64-cpython-311/src/cchardet
      creating build/temp.linux-aarch64-cpython-311/src/ext
      creating build/temp.linux-aarch64-cpython-311/src/ext/uchardet
      creating build/temp.linux-aarch64-cpython-311/src/ext/uchardet/src
      creating build/temp.linux-aarch64-cpython-311/src/ext/uchardet/src/LangModels
      aarch64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -Isrc/ext/uchardet/src -I/home/blitzapi/blitz_api/venv/include -I/usr/include/python3.11 -c src/cchardet/_cchardet.cpp -o build/temp.linux-aarch64-cpython-311/src/cchardet/_cchardet.o
      src/cchardet/_cchardet.cpp:196:12: fatal error: longintrepr.h: No such file or directory
        196 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 1
      [end of output]

>  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

>× Encountered error while trying to install package.
╰─> cchardet

Pi5 runs into these issues in the setup of the rpi-bookworm branch.

**Comment by ostermayer** [comment]
will add 25k sats to this bounty for pi5 support

**Comment by rootzoll** [comment]
The focus of upcomming RaspiBlitz v1.11.0 release is the RaspberryPi5 ... we have a Telegram Group to exchange on RaspberryPi5 community testing: https://t.me/+Ig639m-KD_JjNTQy
